public/API/API_GetGameList.php

```diff
@@ -31,26 +33,50 @@
 
 $withAchievements = (bool) request()->query('f');
 $withHashes = (bool) request()->query('h');
+$offset = (int) request()->query('o');
+$count = (int) request()->query('c');
 
-getGamesListByDev(null, $consoleID, $dataOut, 1, false, $withAchievements ? 0 : 2);
+$query = DB::table('GameData')
+    ->leftjoin('Console AS c', 'c.ID', '=', 'GameData.ConsoleID')
+    ->leftjoin('Achievements AS ach', 'ach.GameID', '=', 'GameData.ID')
+    ->leftjoin('LeaderboardDef AS ld', 'ld.GameID', '=', 'GameData.ID')
+    ->select(
+        'GameData.*',
+        'c.Name as ConsoleName',
+        DB::raw('MAX(ach.DateModified) as DateModified'),
+        DB::raw('COALESCE(GameData.achievements_published,0) AS NumAchievements'),
+        DB::raw('COUNT(DISTINCT ld.ID) AS NumLBs')
+    )
+    ->where('GameData.ConsoleID', $consoleID)
+    ->when($withAchievements, function ($query) {
+        return $query->where('GameData.achievements_published', '>', 0);
+    })
+    ->when($offset > 0 && $count > 0, function ($query) use ($offset) {
+        return $query->offset($offset);
+    })
```

Comment:
The crash is resolved, however this behavior still seems unexpected 🤔

If I only pass an `o` param, I still expect the offset to be respected. At the moment, it's ignored if I don't also provide an accompanying `c` param.

It seems like if I only provide an offset, the limit should be some very large number like 999999.

Reply:
Done. I set a limit of 9999999 when there is no `count`. 

I hope is enough 😅